docs/src/views/ColorPalette.tsx

```diff
@@ -0,0 +1,113 @@
+  <Provider
+    theme={{
+      componentStyles: {
+        ColorBox: colorBoxStyles,
+        ColorVariants: colorVariantsStyles,
+      },
+      componentVariables: {
+        ColorBox: colorBoxVariables,
+      },
+    }}
+  >
+    <ProviderConsumer
+      render={({ siteVariables: { colors, contextualColors, emphasisColors, naturalColors } }) => (
+        <DocPage title="Color Palette">
+          <Header as="h2">Introduction</Header>
+          <p>
+            The color palette for a theme has many requirements and constraints. There is a need to
+            be intentional and functional with color use. We analyzed existing frameworks and picked
+            the best ideas from them.
+          </p>
+          <p>
+            Each theme should match our color palette types fully. This will allow you to use our
+            theming features completely and keep your palette structured.
+          </p>
+
+          <Header as="h2">Primitive colors</Header>
+          <p>
+            This part of the palette includes only <i>black</i> and <i>white</i> colors, we decided
+            to separate by semantical ideas. There is nothing blacker than black and nothing whiter
+            than white.
+          </p>
+
+          <Grid columns={2}>
+            {_.map(['black', 'white'], color => (
+              <Grid.Column key={color}>
+                <ColorBox name={color} rounded size="big" value={colors[color]} />
+              </Grid.Column>
+            ))}
+          </Grid>
+
+          <Header as="h2">Natural colors</Header>
+          <p>
+            This part of palette includes nine colors that are the most frequently used among
+            popular frameworks. Each color includes ten gradients, this allows us to satisfy most
+            common needs. This decision is experienced from Material UI and allows to define more
+            variants than semantical naming (lightest, lighter, etc.).
+          </p>
+
+          <Grid columns={2}>
+            {_.map(naturalColors, (variants, color) => (
+              <Grid.Column key={color}>
+                <ColorBox name={color} rounded value={colors[color][500]} />
+              </Grid.Column>
+            ))}
+          </Grid>
+
+          <Header as="h2">Contextual colors</Header>
+          <p>
+            This part of the palette includes primary and secondary (i.e. brand) colors and
+            contextual colors. Contextual colors can be just aliases for natural colors.
+          </p>
+
+          <Grid columns={2}>
+            {_.map({ ...emphasisColors, ...contextualColors }, (variants, color) => (
+              <Grid.Column key={color}>
+                <ColorBox name={color} rounded size="big" value={colors[color][500]} />
+              </Grid.Column>
+            ))}
+          </Grid>
+
+          <Header as="h2">Text colors</Header>
+          <p>
+            Text variants are also separated as state color because in the most cases it's not
```

Comment:
I am reading this as a client, and have a reasonable question: text color is largely influenced by the backgeround where text is used, but thjis paragraph suggests that there is only single text color suggested (yes, maybe with several tints, but still just a single color) - and this contradicts most approaches to color palette where text color is largely defined by the background - and, thus, `on` variants for text color (generally - content color, such as text, icons, etc) are suggested: `onPrimary`, `onSecondary`, etc. More than that, depending on the tint of the background different text color might be used, so that content will appear with enough visual contrast.

What is our suggestion here?

Reply:
also, might suggest to rename it to something that is more generic than `text`, as this color generally applies not only to text content